client/packages/invoices/src/InboundShipment/DetailView/modals/InboundLineEdit/InboundLineEdit.tsx

```diff
@@ -95,8 +96,15 @@ const useDraftInboundLines = (item: InboundLineItem | null) => {
   );
 
   const saveLines = async () => {
-    if (isDirty) await mutateAsync(draftLines);
-    setIsDirty(false);
+    if (isDirty) {
+      const { errorMessage } = await mutateAsync(draftLines);
+
+      if (errorMessage) {
+        error(errorMessage)();
+      }
```

Comment:
Because this function doesn't throw an error, the modal is closed when an error is shown. 
We keep the modal shown when an error happens, so that the user can correct values without losing all changes - to me it felt odd when the modal closed. I'd suggest this:

```suggestion
      if (errorMessage) {
        throw new Error(errorMessage);
      }
```

as well as fixing the error handler in the button onClick:

```
              } catch (e) {
                error((e as Error).message)();
              }
```

.. I'll make a suggestion branch tho, this comment isn't terribly clear!